src/mscorlib/src/System/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -77,6 +89,19 @@ public static int SizeOf<T>()
         /// </summary>
         [NonVersionable]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void* Add<T>(void* source, int elementOffset)
+        {
+            // The body of this function will be replaced by the EE with unsafe code!!!
+            // See getILIntrinsicImplementationForUnsafe for how this happens.
+            typeof(T).ToString(); // Type token used by the actual method body
+            throw new InvalidOperationException();
```

Comment:
It looks like this isn't being replaced properly.  There are lots of failures in the CI log of the form:
```
               Description: Infinite recursion during resource lookup within System.Private.CoreLib.  This may be a bug in System.Private.CoreLib, or potentially in certain extensibility points such as assembly resolve events or CultureInfo names.  Resource name: Arg_InvalidOperationException
               Stack Trace:
                  at System.SR.InternalGetResourceString(String key)
                  at System.SR.GetResourceString(String resourceKey, String defaultString)
                  at System.InvalidOperationException..ctor()
                  at System.Runtime.CompilerServices.Unsafe.Add[T](T& source, Int32 elementOffset)
                  at System.IO.UnmanagedMemoryStream.Read(Byte[] buffer, Int32 offset, Int32 count)
                  at System.IO.BinaryReader.FillBuffer(Int32 numBytes)
                  at System.IO.BinaryReader.ReadInt32()
                  at System.Resources.ManifestBasedResourceGroveler.CreateResourceSet(Stream store, Assembly assembly)
                  at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark)
                  at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark)
                  at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
                  at System.Resources.ResourceManager.GetString(String name, CultureInfo culture)
                  at System.SR.InternalGetResourceString(String key)
                  at System.SR.GetResourceString(String resourceKey, String defaultString)
                  at System.InvalidOperationException..ctor()
                  at System.Runtime.CompilerServices.Unsafe.Add[T](T& source, Int32 elementOffset)
                  at System.IO.UnmanagedMemoryStream.Read(Byte[] buffer, Int32 offset, Int32 count)
                  at System.IO.BinaryReader.FillBuffer(Int32 numBytes)
                  at System.IO.BinaryReader.ReadInt32()
                  at System.Resources.ManifestBasedResourceGroveler.CreateResourceSet(Stream store, Assembly assembly)
                  at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark)
                  at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark)
                  at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
                  at System.Resources.ResourceManager.GetString(String name, CultureInfo culture)
                  at System.SR.InternalGetResourceString(String key)
                  at System.SR.GetResourceString(String resourceKey, String defaultString)
                  at System.NullReferenceException..ctor()
                  at System.Runtime.InteropServices.Marshal.ReadInt64(IntPtr ptr, Int32 ofs)
                  at ReadWriteInt64Test.NullValueTests()
                  at ReadWriteInt64Test.Main(String[] unusedArgs)
```

Reply:
I don't think I defined the method in vm correctly (was following changes from previous changes like this https://github.com/dotnet/coreclr/pull/9808) but I am not sure how to correct it.
@jkotas, can you help with this?

I added a METASIG for the PTR_ADD method because I wasn't sure how the two methods could be differentiated if I left it as NoSig.
DEFINE_METHOD(UNSAFE,               BYREF_ADD,              Add, NoSig)
DEFINE_METHOD(UNSAFE,               PTR_ADD,                Add, GM_PtrVoid_Int_RetPtrVoid)

Reply:
You cannot use NoSig anymore for any of the Add methods because of it is ambiguous match. All overloads have to have the signature specified now.

Reply:
I specified the signature for the Add and As methods.
However, I am getting the following build errors. Any ideas on how to resolve this?

Assert failure(PID 8012 [0x00001f4c], Thread: 24648 [0x6048]): Compiler optimization assumption invalid: EE expects method to exist: System.Runtime.CompilerServices.Unsafe:As  Sig pointer: 00007FF7A8E69CC0
FAILED: pMD != 0

CROSSGEN! CHECK::Trigger + 0x254 (0x00007ff7`a8b08894)
CROSSGEN! MscorlibBinder::LookupMethodLocal + 0x1EF (0x00007ff7`a886963f)
CROSSGEN! MscorlibBinder::GetMethodLocal + 0x4A (0x00007ff7`a8866e4a)
CROSSGEN! MscorlibBinder::BindAll + 0x89 (0x00007ff7`a88662a9)
CROSSGEN! Module::ExpandAll + 0x105A (0x00007ff7`a87ad99a)
CROSSGEN! CEEPreloader::Preload + 0x8A (0x00007ff7`a86fb89a)
CROSSGEN! CEECompileInfo::PreloadModule + 0xA1 (0x00007ff7`a86fb981)
CROSSGEN! ZapImage::Preload + 0x9C (0x00007ff7`a8aa8d0c)
CROSSGEN! Zapper::CompileModule + 0x190 (0x00007ff7`a8a85890)
CROSSGEN! Zapper::CompileAssembly + 0x3BD (0x00007ff7`a8a84dfd)
    File: d:\github\fork\coreclr\src\vm\binder.cpp Line: 128
    Image: D:\GitHub\Fork\coreclr\bin\Product\Windows_NT.x64.Debug\crossgen.exe

Reply:
`Unsafe.As(ref` signature is not `RefT_RetRefT`. It is `RefTFrom_RetRefTTo`.

Reply:
I updated the signature. I see the same error still.

Reply:
> Should the 3rd argument be 2 instead of 1?

That fixed the error! :)